<h2>Soal 1: Membuat Objek DataFrames</h2>

- Buatlah objek dataframes dari objek dictionary, Bebas!

In [8]:
# Membuat dataframes dari dictionary
import pandas as pd

pd.DataFrame({"Tinggi Siswa" : [150,165,170,155,167], "Nama Siswa" : ["Cindy","Andi","Tamara","Budi","Jamal"]})

Tinggi Siswa Nama Siswa
0           150      Cindy
1           165       Andi
2           170     Tamara
3           155       Budi
4           167      Jamal

<h2>Soal 2: Membaca data dari file</h2>

Download file bernama [diamonds.csv](https://drive.google.com/uc?export=download&id=1qQiM9utQUThTPq8jk8stPMh_y2zfqYOd), kemudian baca data tersebut dan assign ke dalam suatu variabel kemudian print.

In [9]:
# Membaca data dan memasukannya kedalam suatu variabel
df = pd.read_csv('diamonds-1.csv')
df

Unnamed: 0  carat        cut color clarity  depth  table  price     x   
0               1   0.23      Ideal     E     SI2   61.5   55.0    326  3.95  \
1               2   0.21    Premium     E     SI1   59.8   61.0    326  3.89   
2               3   0.23       Good     E     VS1   56.9   65.0    327  4.05   
3               4   0.29    Premium     I     VS2   62.4   58.0    334  4.20   
4               5   0.31       Good     J     SI2   63.3   58.0    335  4.34   
...           ...    ...        ...   ...     ...    ...    ...    ...   ...   
53935       53936   0.72      Ideal     D     SI1   60.8   57.0   2757  5.75   
53936       53937   0.72       Good     D     SI1   63.1   55.0   2757  5.69   
53937       53938   0.70  Very Good     D     SI1   62.8   60.0   2757  5.66   
53938       53939   0.86    Premium     H     SI2   61.0   58.0   2757  6.15   
53939       53940   0.75      Ideal     D     SI2   62.2   55.0   2757  5.83   

          y     z  
0      3.98  2.43  
1      3.84  2.31  
2      4.07  2.31  
3      4.23  2.63  
4      4.35  2.75  
...     ...   ...  
53935  5.76  3.50  
53936  5.75  3.61  
53937  5.68  3.56  
53938  6.12  3.74  
53939  5.87  3.64  

[53940 rows x 11 columns]

Expected Result:

![alt text](https://drive.google.com/uc?id=1BvgM5mJEVfj0fllrjkBfQwn0QIFkry_t)

<h2>Soal 3: DataFrame vs Series</h2>

- Apa perbedaan Series dengan list dan dictionary
- Apa perbedaan dataframe dan series
- Buatlah suatu data series

Jawab:
- Perbedaan series dengan list dan dictionary adalah pada tipe data dan dimensinya, Data value pada Series harus bertipe sama, sedangkan pada List dan Dictionary pada value dapat berbeda-beda tipe datanya. Objek Series hanya bersifat satu dimensi saja.

- Perbedaan dataframe dan series adalah terletak pada struktur, atribute, dan method-method yang dimiliki masing2
untuk Series memiliki objek 1 dimensi, sedangkan Dataframe memiliki objek 2 dimensi mirip seperti table.


In [10]:
# code here
print(pd.Series([6,8,9,0]))

0    6
1    8
2    9
3    0
dtype: int64


<h2>Soal 4: Pemeriksaan Data Sederhana</h2>

Di soal no 2 kamu telah membaca data tentang diamonds. pada soal ini cobalah suatu metode dari pandas untuk mengetahui beberapa data pertama dan beberapa data terakhir. kemudian sebutkan nilai dari:
- baris pertama dari column price
- baris terakhir dari column color

tunjukan cara kalian menemukan nilai tersebut di block di bawah ini.

In [11]:
# code here
print("Nilai baris pertama column Price :",df.price.iloc[0])
print("Nilai baris terakhir column color :",df.color.iloc[-1])

Nilai baris pertama column Price : 326
Nilai baris terakhir column color : D


<h2>Soal 5: Deskriptif Statistik</h2>

Jelaskan apa itu deskriptis statistik!

Jawab: Salah satu bagian dari ilmu statistika yang berhubungan dengan aktivitas penghimpunan, penataan, peringkasan dan penyajian data dengan harapan agar data lebih bermakna, mudah dibaca dan mudah dipahami oleh pengguna data meliputi pengumpulan, pengelompokan dan pengolahan data yang selanjutnya akan menghasilkan ukuran-ukuran statistik seperti frekuensi, pemusatan data, penyebaran data, kecenderungan suatu gugus data dan lain-lain.

<h2>Soal 6: Practice Deskriptif Statistik</h2>

- Gunakan suatu metode dari pandas untuk mengetahui deskriptif statistik dari suatu data
- Berapa nilai rata-rata dari column price
- Berapa nilai standar deviasi dari column depth
- Berapa nilai maximum dari column carat

In [12]:
# code here
df.describe()

Unnamed: 0         carat         depth         table         price   
count  53940.000000  53940.000000  53940.000000  53940.000000  53940.000000  \
mean   26970.500000      0.797940     61.749405     57.457184   3932.799722   
std    15571.281097      0.474011      1.432621      2.234491   3989.439738   
min        1.000000      0.200000     43.000000     43.000000    326.000000   
25%    13485.750000      0.400000     61.000000     56.000000    950.000000   
50%    26970.500000      0.700000     61.800000     57.000000   2401.000000   
75%    40455.250000      1.040000     62.500000     59.000000   5324.250000   
max    53940.000000      5.010000     79.000000     95.000000  18823.000000   

                  x             y             z  
count  53940.000000  53940.000000  53940.000000  
mean       5.731157      5.734526      3.538734  
std        1.121761      1.142135      0.705699  
min        0.000000      0.000000      0.000000  
25%        4.710000      4.720000      2.910000  
50%        5.700000      5.710000      3.530000  
75%        6.540000      6.540000      4.040000  
max       10.740000     58.900000     31.800000

In [13]:
print(df.price.mean())
print(df.depth.std())
print(df.carat.max())

3932.799721913237
1.432621318833661
5.01


Jawab:

- Nilai rata-rata dari column price : 3932.799722
- Nilai standar deviasi column depth : 1.432621
- Nilai maximum column carat : 5.010000